test/e2e/providers/aws/aws.go

```diff
@@ -114,18 +114,33 @@ func (a *awsProvider) getInfraID() (string, error) {
 	return infraID, nil
 }
 
+// getWindowsAMIFilter returns a ready to use EC2 AMI filter for the Windows Server image.
+// This filter will grab all AMI's that match the exact name. The '?' indicate any character will match.
+// For example, the AMI's will have the name format: Windows_Server-2022-English-Full-Base-2022.06.15
+// so the question marks will match the date of creation.
+func getWindowsAMIFilter() string {
+	// The Windows Server version can be provided as an environment variable. For example, from the CI job definition
+	// in the release step. The selected Windows Server version must be compatible with Windows container image.
+	// See https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/base-image-lifecycle
+	windowsServerVersion := os.Getenv("WINDOWS_SERVER_VERSION")
```

Comment:
please make this check within `SetupAWSCloudProvider` and make it part of the awsProvider struct, for the same reasons as above

Reply:
In contrast with above case, the cardinality of the  `SetupAWSCloudProvider` func with the proposed `getWindowsAMIFilter` func is one to one. So that, the syscall to read the env var (`os.Getenv()`) is executed just one time in the execution flow.  IMO, the `windowsServerVersion` field is specific to the e2e test and should not be part of the `awsProvider` struct.